src/Microsoft.AspNetCore.Mvc.Formatters.Json/Internal/JsonArrayPool.cs

```diff
@@ -33,7 +39,11 @@ public void Return(T[] array)
                 throw new ArgumentNullException(nameof(array));
             }
 
-            _inner.Return(array);
+            if (_arrayTracker.Contains(array))
+            {
+                _arrayTracker.Remove(array);
```

Comment:
Can combine `Contains()` and `Remove()` calls: `if (_arrayTracker.Remove(array))`